Amazon.Ion.ObjectMapper/IonDictionarySerializer.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Amazon.IonDotnet;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Amazon.Ion.ObjectMapper
+{
+    public class IonDictionarySerializer : IonSerializer<IDictionary>
+    {
+        private IonSerializer serializer;
+        private Type valueType;
+
+        public IonDictionarySerializer(IonSerializer ionSerializer, Type valueType)
+        {
+            this.serializer = ionSerializer;
+            this.valueType = valueType;
+        }
+
+        public override IDictionary Deserialize(IIonReader reader)
```

Comment:
Is it not possible to have a generic IDictionary? Like IDictionary<String, T> ?

Reply:
No, it is not possible.

Reply:
Why?

Reply:
Because the return value of Deserialize must match the generic type of the IonSerializer interface implemented.

We cannot have a generic type of `IonSerializer<string, T>` without changing the clsas to `IonDictionarySerializer<T>`, but we do not have context of what `T` is from `IonSerializer.Serialize` since at that point it has been reduced to `typeof(T)`.